kube-runtime/src/events.rs

```diff
@@ -291,15 +396,18 @@ mod test {
 
         let svcs: Api<ClusterRole> = Api::all(client.clone());
         let s = svcs.get("system:basic-user").await?; // always get this default ClusterRole
-        let recorder = Recorder::new(client.clone(), "kube".into(), s.object_ref(&()));
+        let recorder = Recorder::new(client.clone(), "kube".into());
         recorder
-            .publish(Event {
-                type_: EventType::Normal,
-                reason: "VeryCoolServiceNoNamespace".into(),
-                note: Some("Sending kubernetes to detention without namespace".into()),
-                action: "Test event - plz ignore".into(),
-                secondary: None,
-            })
+            .publish(
+                Event {
+                    type_: EventType::Normal,
+                    reason: "VeryCoolServiceNoNamespace".into(),
+                    note: Some("Sending kubernetes to detention without namespace".into()),
+                    action: "Test event - plz ignore".into(),
+                    secondary: None,
+                },
+                &s.object_ref(&()),
+            )
```

Comment:
if we are changing .publish, we should make the integration test also publish twice to the same name and verify we get an EventSeries

Reply:
Agree. I will try to add them soon.

Reply:
Fixed on 07145abc2afaf345fdee0e8195180c1b299b09b3

Reply:
I don't think this is actually tested still. We should have an actual event series, with a `count` of 2 if we publish twice, and i tried this in controller-rs with this branch and am not seeing event aggregation from kubectl describe.

maybe i am doing something wrong though. have outlined my steps in https://github.com/kube-rs/controller-rs/pull/116 and set it up against your branch

Reply:
ah. i'm guessing this is because the event key contains an `ObjectReference` for equality, and objectref contains a `resourceVersion` meaning the ones i am changing are actually "not the same object", and would not be aggregated.

not sure how i feel about that. in my mind the aggregation ought to be unique per GVK, but maybe that's not the standard way of doing it in kubernetes land 🤔

Reply:
Indeed we are not using `resourceVersion` in Hash. 
```
impl Hash for Reference {
    fn hash<H: Hasher>(&self, state: &mut H) {
        self.0.api_version.hash(state);
        self.0.kind.hash(state);
        self.0.name.hash(state);
        self.0.namespace.hash(state);
        self.0.uid.hash(state);
    }
}
```

I checked your code and saw that the recorder is initialized each reconcile cycle, so your cache is empty. I initialized it [in the context creation](https://github.com/pando85/kaniop/blob/9d3fa6ad695733e5faf9d0c7b39e1f8eb1a16cbe/libs/operator/src/controller.rs#L161).

On 07145abc2afaf345fdee0e8195180c1b299b09b3 we are testing to send it twice and check that `series` exists. We could add more checks or cases if needed.

Reply:
ahhh. that explains the changing of the oref argument as well. changing this to create it on the context creates the correct results;
```
  Normal  HideRequested  56s (x2 over 75s)  doc-controller  Hiding `samuel`
```